File: backend/app/schemas/water.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class WaterLogCreate(BaseModel):
    amount_ml: int = Field(gt=0, le=5000)
    logged_at: datetime | None = None


class WaterLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    amount_ml: int
    logged_at: datetime


class WaterSummary(BaseModel):
    goal_ml: int
    total_ml_today: int
    logs_today: list[WaterLogRead]

File: mobile/src/api/water.ts
import { api } from "./client";

export type WaterLog = {
  id: number;
  amount_ml: number;
  logged_at: string;
};

export type WaterSummary = {
  goal_ml: number;
  total_ml_today: number;
  logs_today: WaterLog[];
};

export async function getTodayWaterSummary(): Promise<WaterSummary> {
  const { data } = await api.get<WaterSummary>("/water/today");
  return data;
}

export async function logWater(amountMl: number): Promise<WaterLog> {
  const { data } = await api.post<WaterLog>("/water", { amount_ml: amountMl });
  return data;
}
